flink-runtime/src/main/java/org/apache/flink/runtime/util/config/memory/ManagedMemoryUtils.java

```diff
@@ -155,4 +155,19 @@ public static double getFractionRoundedDown(final long dividend, final long divi
                         BigDecimal.ROUND_DOWN)
                 .doubleValue();
     }
+
+    public static void validateManagedMemoryUseCaseWeights(
+            Map<ManagedMemoryUseCase, Integer> existingOperatorScopeUseCaseWeights,
+            Map<ManagedMemoryUseCase, Integer> newOperatorScopeUseCaseWeights) {
```

Comment:
It's a bit unclear from the method name what does the method validate.
I'd suggest `validateUseCaseWeightsNoConflict(weights1, weights2)`.